### ISSUE TITLE
Share link should share bare URL not magic login token

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#5558]
**Describe the bug**
On a phone, if you click through from an email notification to an expense, then use the browser 'share' button, it will share the link with the secure login token from the email instead of the bare URL. Might be due to how gmail on android opens links in its own browser instead of kicking it out into the phone's main browser app? Is this a security risk?

**To Reproduce**

1. Click the view button on an expense email notification
2. Link opens in gmail's internal browser
3. Click share and send the URL to someone over a chat app
4. URL is full string with login token included

**Expected behavior**
Share bare version of the expense URL

**Screenshots**

![signal-2022-05-14-125226](https://user-images.githubusercontent.com/5498878/168405326-6e17831d-c636-4526-9739-f5dc0b37997b.jpeg)

![signal-2022-05-14-125352](https://user-images.githubusercontent.com/5498878/168405325-5ed34fb8-7683-4990-a99c-d61deb18d6ff.jpeg)

![](https://user-images.githubusercontent.com/5498878/168405321-12bf0945-4fe1-4f12-922f-77e6042a7b12.png)

![](https://user-images.githubusercontent.com/5498878/168405434-08339d99-2f70-414f-a720-fcb78920216b.png)
token string redacted but it shared the full string

**Desktop (please complete the following information):**
 - OS: Android
 - Browser: Gmail integrated browser

**Additional context**

  P2 low frequency, high impact

**Comment by Betree** [comment]
@alanna I just tried on my own Android phone with GMail / integrated browser activated and did not get this issue. Could it be that you clicked on share before the URL redirect happened? Are you running on the latest GMail app version?

**Comment by znarf** [comment]
Notification emails are not containing any sign in token. It's possible there is a step missing where the user arrive on unauthenticated page then has to go through the normal sign in flow.

Also, I'll note that a sign in token only works once and is invalidated as soon as it's used.

**Comment by mixmix** [comment]
hey sorry I don't know the exact path I took. I was trying to approve an expense on OC, and had received an email I think? Or maybe I went to OC and asked for a magic link. I was able to produce it twice - once when Alanna was like "what did you do!" the second time where I was moving slower to take screenshots.

It seems to me you have small vulnerability that needs closing. It will depend a lot on how your magic links work - if they are single use you are probably fine... I think instead you might have built them so that you can use them N times with M minutes?

Can give you versions:
- Gmail - version `2022.04.16.44632206.Release`

**Comment by Betree** [comment]
> I think instead you might have built them so that you can use them N times with M minutes?

@mixmix No, as @znarf mentioned above they're single-use. If you're logged in already, you'll be redirected and won't see any error while trying to re-use the link. But if you're not, you'll see an error message saying that your token has expired.

**Comment by alanna** [comment]
OK, then it seems the token gets shared somehow with the URL in some cases,  but that's not a big deal because it's one time use. Still odd for the user to see but not a security risk.

**Comment by znarf** [comment]
I tried to reproduce, and wondered from where Gmail would decide to share that URL. I haven't find clues on our side.

On Gmail side, that would also look like an important bug (URL in navigation being different from URL shared). So, root cause is still unclear to me.
